crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -275,6 +275,31 @@ use std::{cell::UnsafeCell, marker::PhantomData};
 /// # bevy_ecs::system::assert_is_system(my_system);
 /// ```
 ///
+/// ## Non-Sync Queries
+///
+/// Non-`Sync` components are supported, but cannot be accessed as `&T`,
+/// as the type cannot be safely read from multiple threads at the same
+/// time. These components must be accessed via a `&mut T` to guarentee
+/// that the system has exclusive access to the components.
```

Comment:
this isnt going to work and imo is flawed anyway, why shouldn't you be able to use `&T` from multiple places in a system? 
imo we should do one of:
- Add  `NonSync` versions of `&T` and `&mut T`, leaving the existing types requiring `T: Sync`
- Remove `T: Sync` bound from `&T` and `&mut T` worldqueries (this is a bad idea since now we're going to be registering almost eveyr component as `!Sync`)
- Add a `NonSync<T>` and use that as the readonly version of `&mut T` (this is a bad idea since its going to screw up the `Query::to_readonly` effort by making the readonly version of `Query<&mut T>` not be `Query<&T>`)

I am assuming here that the scheduler is smart enough to know not to schedule two systems with read only access of a `!Sync` component in parallel, if not that should be fixed

Reply:
Why not just add `NonSyncRead<T>` for the 'use &T from multiple places in a system' case?

Fundamentally, `&mut T` requiring `T: Sync` is entirely superfluous - the requirement that `T` is `Sync` would never be used.

Reply:
Oh I see, because we need to convert to readonly. That's annoying.
I don't suppose we can gate a readonly conversion existing on T being `Sync`? Is conversion to readonly really a fundamental part of the Query API?

Reply:
conversion to readonly doesnt _require_ it. we could after all set the readonly version of `&mut T` to be `NonSync<T>`. It's just kinda silly to have the readonly version of `&mut T` not be `&T`.

> Is conversion to readonly really a fundamental part of the Query API?

I think so? It's not quite there yet but I hope that in a bit of time we'll have much more flexibility around changing the `Q/F` generics and being able to downgrade a query to readonly is a part of that. Personally I value `to_readonly` working on everything a lot more than some extreme edge case of non sync components needing to use `NonSyncMut<T>` instead of `&mut T`